k8s/k8s.go

```diff
@@ -171,13 +171,22 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 			}
 		}
 
-		// Run the IPAM plugin
-		logger.Debugf("Calling IPAM plugin %s", conf.IPAM.Type)
-		result, err = ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
-		if err != nil {
-			return nil, err
+		ipAddrsNoIpam := annot["cni.projectcalico.org/ipAddrsNoIpam"]
+
+		// Call IPAM plugin if ipAddrsNoIpam annotation is not present.
+		if ipAddrsNoIpam == "" {
+			// Run the IPAM plugin
+			logger.Debugf("Calling IPAM plugin %s", conf.IPAM.Type)
+			result, err = ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
+			if err != nil {
+				return nil, err
+			}
+			logger.Debugf("IPAM plugin returned: %+v", result)
+		} else {
```

Comment:
(discussed this offline. Going to leave this as it is for now. Will need to refactor how we call k8s APIs at some point.